### DISCUSSION TITLE
Module M test fails

### DISCUSSION THREAD

**Discussion by MaximePrs** [neperfepx/neper#541] [Q&A] [answered]
Hello,

I have some issues while running `make test` and I don't know how to solve it.
It seems similar to the following discussion [Neper tests FAILED](https://github.com/neperfepx/neper/discussions/443).
Even when making the indicated corrections, the problem is not solved.

Neper version: 4.4.2-33
Gmsh version: 4.11.0 (available with command line gmsh)
Ubuntu version: 18.04
Failed test list:
```
88% tests passed, 40 tests failed out of 331

Total Test time (real) = 242.72 sec

The following tests FAILED:
	128 - M/cl (Failed)
	129 - M/cl_expr (Failed)
	132 - M/clmin (Failed)
	133 - M/clratio (Failed)
	134 - M/cut (Failed)
	135 - M/dim (Failed)
	136 - M/dim2 (Failed)
	137 - M/dim_expr (Failed)
	138 - M/faset (Failed)
	141 - M/interface (Failed)
	142 - M/interface2 (Failed)
	143 - M/interface3 (Failed)
	144 - M/mesh2dalgo_dela (Failed)
	145 - M/mesh2dalgo_fron (Failed)
	146 - M/mesh2dalgo_mead (Failed)
	147 - M/mesh2dalgo_netg (Failed)
	148 - M/mesh3dalgo_netggmne (Failed)
	149 - M/mesh3dalgo_netggmsh (Failed)
	150 - M/mesh3dalgo_netgnetg (Failed)
	151 - M/meshing (Failed)
	153 - M/meshqualdisexpr (Failed)
	154 - M/meshqualexpr (Failed)
	155 - M/meshqualmin (Failed)
	156 - M/morpho_cube (Failed)
	157 - M/nset1 (Failed)
	158 - M/nset2 (Failed)
	159 - M/nset3 (Failed)
	160 - M/order (Failed)
	161 - M/order_dim2 (Failed)
	166 - M/part_dim2 (Failed)
	167 - M/pl (Failed)
	168 - M/rcl (Failed)
	169 - M/rcl_expr (Failed)
	172 - M/remesh1 (Failed)
	173 - M/remesh2 (Failed)
	174 - M/remesh3 (Failed)
	180 - M/statelsetco (Failed)
	186 - M/stateltco (Failed)
	188 - M/tesr_dim2 (Failed)
	191 - M/tess2ddim3 (Failed)
Errors while running CTest
Output from these tests are in: /home/dmsm/m.piras/Documents/neper/src/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
Makefile:90: recipe for target 'test' failed
make: *** [test] Error 8
```

Do you know if there is a solution or could I use neper without solving it ?

Many thanks.

**Comment by rquey** [comment, marked as answer]
Gmsh installation issue. 

Since version 4.11 is still under development, I assume you have compiled it from the sources. Did you resolve all dependencies (which you can see at Gmsh's `cmake ..` stage)?

I just installed the last Gmsh (development) version, and all Neper tests pass on my computer.

**Reply by MaximePrs** [reply]
I tried to solve all the dependencies issues with the IT service to install correctly Gmsh (with stable release 4.10.5) and then rebuilt neper.
After those steps, the problem is not solved. I even have more failed tests (from module T and M and 1 from module V).
However, when I run the minimal test mode, everything pass.
Is the problem from the different versions of software ?

Could I assume that neper works correctly because minimal tests pass ?

**Reply by rquey** [reply]
Yes, it's a good sign.  Not sure what is happening, but it may be only that the reference results (which normal testing compares against) were built on Ubuntu 20.04 while you are running 18.04...

**Reply by MaximePrs** [reply]
The IT service also tried to install Neper using a virtual machine with Ubuntu 20.04 and all the dependencies with the version (or higher) used to generate the output files.
It is the same result observed when running the tests for me; there are many failed tests (174).

Since tests passed when using minimal mode, I will use Neper as it is and hope it will not induce too many errors.
Thank you for your support.

**Reply by l-matSci** [reply]
Hello Romain, 

I have a similar issue, and thus replying here (should i make a new thread?)
I am trying an installation on WSL, with Ubuntu 22.04.
The tests are failing mostly in the M and V modules. I have installed the dependencies (GNU Scientific Library, OpenMP, pthread, Gmsh 4.8.4, POV-Ray and Asymptote).

Any Ideas what i am doing wrong? My assumption is either gmsh or asymptote, but i cannot locate the issue.

128:M/cl
129:M/cl_expr
132:M/clmin
133:M/clratio
134:M/cut
135:M/dim
136:M/dim2
137:M/dim_expr
138:M/faset
141:M/interface
142:M/interface2
143:M/interface3
144:M/mesh2dalgo_dela
145:M/mesh2dalgo_fron
146:M/mesh2dalgo_mead
147:M/mesh2dalgo_netg
148:M/mesh3dalgo_netggmne
149:M/mesh3dalgo_netggmsh
150:M/mesh3dalgo_netgnetg
151:M/meshing
153:M/meshqualdisexpr
154:M/meshqualexpr
155:M/meshqualmin
156:M/morpho_cube
157:M/nset1
158:M/nset2
159:M/nset3
160:M/order
161:M/order_dim2
166:M/part_dim2
167:M/pl
168:M/rcl
169:M/rcl_expr
172:M/remesh1
173:M/remesh2
174:M/remesh3
175:M/remesh4
181:M/statelsetco
187:M/stateltco
189:M/tesr_dim2
192:M/tess2ddim3
197:V/cameraangle
198:V/cameracoo1
199:V/cameracoo2
200:V/cameracoo3
201:V/cameralookat1
202:V/cameralookat2
203:V/cameralookat3
204:V/cameraprojection
205:V/camerasky
206:V/cellcol_id
207:V/cellcol_ori
208:V/cellcol_real
209:V/cellcol_real_colscheme
210:V/cellcol_rgb
211:V/cellcol_rgb2
212:V/cellscale
213:V/cellscale2
214:V/cellscaletitle
215:V/celltrs
216:V/crystalcol_colscheme
217:V/crystalcol_id
218:V/crystalcol_id_cubic
219:V/crystalcol_id_hexagonal
220:V/crystalcol_ori
221:V/crystalcol_per
222:V/crystalcol_real
223:V/crystalcol_real_colscheme
224:V/crystalcol_rgb
225:V/crystalcol_rgb2
226:V/crystaledgecol_cubic
227:V/crystaledgerad_cubic
228:V/crystalscale
229:V/crystalscale2
230:V/crystalscaletitle
231:V/crystaltrs
232:V/csys
233:V/csyscol
234:V/csyscoo
235:V/csyslabel
236:V/csyslength
237:V/csysrad
238:V/custom_input
239:V/custom_input_col
240:V/custom_input_multiple
241:V/custom_input_multiple_col
242:V/custom_input_multiple_rad
243:V/custom_input_multiple_show
244:V/custom_input_multiple_show2
245:V/custom_input_rad
246:V/dataelt3d
247:V/dataexpr
248:V/datasimtesr
249:V/datasimtess
250:V/ebsd1
251:V/ebsd10
252:V/ebsd11
253:V/ebsd2
254:V/ebsd3
255:V/ebsd4
256:V/ebsd5
257:V/ebsd6
258:V/ebsd7
259:V/ebsd8
260:V/ebsd9
261:V/edgecol_id
262:V/edgecol_real
263:V/edgecol_real_colscheme
264:V/edgecol_rgb
265:V/edgecol_rgb2
266:V/edgescale
267:V/edgescale2
268:V/edgescaletitle
269:V/edgetrs
270:V/elsetcol
271:V/elsetcol_ori
272:V/elsetcol_ori_slice
273:V/elsetcol_real
274:V/elsetcolscheme
275:V/elsetscale
276:V/elsetscale2
277:V/elsetscaletitle
278:V/facecol_id
279:V/facecol_real
280:V/facecol_real_colscheme
281:V/facecol_rgb
282:V/facecol_rgb2
283:V/facescale
284:V/facescale2
285:V/facescaletitle
286:V/facetrs
287:V/imagebackground
289:V/imagesize
290:V/loop
291:V/nodecol
292:V/nodecol_real
293:V/nodecoo
294:V/nodecoo_disp
295:V/nodecoofact
296:V/nodescale
297:V/nodescale2
298:V/polycol_id
299:V/polycol_real
300:V/polycol_real_colscheme
301:V/polycol_rgb
302:V/polycol_rgb2
303:V/polyscale
304:V/polyscale2
305:V/polyscaletitle
306:V/polytrs
307:V/povrayantialiasing
308:V/seedcol_id
309:V/seedcol_per
310:V/seedcol_real
311:V/seedcol_real_colscheme
312:V/seedcol_rgb
313:V/seedcol_rgb2
314:V/seedscale
315:V/seedscale2
316:V/seedscaletitle
317:V/seedtrs
318:V/show
319:V/showelt
320:V/showfaceinter
321:V/showmesh1
322:V/showmesh2
323:V/slicemesh
324:V/slicemesh2
325:V/slicemesh3
326:V/slicemesh4
327:V/slicemesh5
328:V/slicemesh6
329:V/space_pf
330:V/space_pf_2d
331:V/space_pf_2d_tesr
332:V/space_pf_cellcol
333:V/space_pf_cellcol_tesr
334:V/space_pf_cellrad
335:V/space_pf_cellrad_tesr
336:V/space_pf_cellsymbol_diamond
337:V/space_pf_cellsymbol_diamond_tesr
338:V/space_pf_cellsymbol_square
339:V/space_pf_cellsymbol_square_tesr
340:V/space_pf_cellsymbol_star
341:V/space_pf_cellsymbol_star_tesr
342:V/space_pf_cellsymbol_triangle
343:V/space_pf_cellsymbol_triangle_tesr
344:V/space_pf_clustering
345:V/space_pf_custom_input
346:V/space_pf_custom_input_ori
347:V/space_pf_custom_input_ori_des
348:V/space_pf_custom_input_vect
349:V/space_pf_custom_input_vect_pfdir
350:V/space_pf_custom_input_vect_pfdir2
351:V/space_pf_density
352:V/space_pf_density_pfdir
353:V/space_pf_density_pfdir2
354:V/space_pf_densitycolscheme
355:V/space_pf_densitygridsize
356:V/space_pf_densitykernel
357:V/space_pf_densitypoint
358:V/space_pf_densityscale
359:V/space_pf_densityscaleticks
360:V/space_pf_pfdir
361:V/space_pf_pfdir2
362:V/space_pf_pffont
363:V/space_pf_pfpole
364:V/space_pf_pfproj
365:V/space_pf_sym
366:V/space_pf_tesr
367:V/space_tree
368:V/space_tree_res
369:V/tesr
370:V/tesr2d
371:V/vercol_id
372:V/vercol_real
373:V/vercol_real_colscheme
374:V/vercol_rgb
375:V/vercol_rgb2
376:V/verrad
377:V/verscale
378:V/verscale2
379:V/verscaletitle
380:V/vertrs

**Comment by eugenianietov** [comment]
Hi all!

I just had the same issue when installing Neper in Ubuntu 20.04 and I fixed it by installing the latest version of Gmsh, which is 4.11. Now all tests pass. When I installed Gmsh from terminal (apt-get), the version by default is 4.4, which apparently fails for Neper tests. I hope this helps!

Best,
Eugenia

**Reply by mkasemer** [reply]
To this comment (@rquey can correct me if I am wrong), I think there are a couple of points to be made:
- Overall, you are much better off installing Gmsh from source rather than via `apt-get`, as the latter will have a severe drop in efficiency. The overhead of source installation is worth it.
- I have also experienced some issues with Gmsh on Ubuntu 22 for meshing. The tests fail, but not because the meshes are different: only because some of the ordering in the mesh file is different. So, the difference that the testing is looking for between the reference mesh and the printed mesh yields a failure, but they are---for practical purposes---the same. To @eugenianietov, I think you can use a newer version of Gmsh no problem, just dig into the meshing failures to see that when interpreted, they are likely yielding practically the same results.

**Comment by kight55** [comment]
hi
Where can I find the installation package for 4.11?
use  apt-get is 4.8.4.

**Reply by rquey** [reply]
As @mkasemer pointed out above, "*you are much better off installing Gmsh from source rather than via apt-get, as the latter will have a severe drop in efficiency. The overhead of source installation is worth it.*".

https://gmsh.info
